packages/account/src/account.test.ts

```diff
@@ -899,7 +899,7 @@ describe('Account', () => {
     vi.restoreAllMocks();
   });
 
-  it('should validate max number of inputs when funding the TX', async () => {
+  it('throws when funding with more than 255 coins for an input', async () => {
```

Comment:
`getCoinsToSpend` query will throw when requesting more than 255 coins as documented [here](https://github.com/FuelLabs/fuel-core/issues/2358#issuecomment-2413126236). I've updated this case to reflect that.

Reply:
Should we move our [validation](https://github.com/FuelLabs/fuels-ts/blob/aa/chore/fuel-core-0.40.0/packages/account/src/providers/provider.ts#L846) to [before](https://github.com/FuelLabs/fuels-ts/blob/aa/chore/fuel-core-0.40.0/packages/account/src/providers/provider.ts#L1469) the `getCoinsToSpend` request so that this error is more informative?

Some changes to the validation method might be required.

Reply:
We need to validate the request once inputs have been added, therefore after the `getCoinsToSpend` call. I thought about putting it inside `getCoinsToSpend` but it has no awareness of the request.